Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp

```diff
@@ -462,25 +462,47 @@ void BlockFormattingContext::resolve_used_height_if_treated_as_auto(Box const& b
         height = max(height, calculate_inner_height(box, available_space.height, computed_values.min_height()));
     }
 
+    auto writing_mode = box.computed_values().writing_mode();
+    bool has_horizontal_writing_mode = (writing_mode == CSS::WritingMode::HorizontalTb);
+
     if (box.document().in_quirks_mode()
         && box.dom_node()
         && box.dom_node()->is_html_html_element()
-        && box.computed_values().height().is_auto()) {
+        && ((box.computed_values().width().is_auto() && !has_horizontal_writing_mode)
+            || (box.computed_values().height().is_auto() && has_horizontal_writing_mode))) {
         // 3.6. The html element fills the viewport quirk
         // https://quirks.spec.whatwg.org/#the-html-element-fills-the-viewport-quirk
-        // FIXME: Handle vertical writing mode.
 
         // 1. Let margins be sum of the used values of the margin-left and margin-right properties of element
         //    if element has a vertical writing mode, otherwise let margins be the sum of the used values of
         //    the margin-top and margin-bottom properties of element.
-        auto margins = box_state.margin_top + box_state.margin_bottom;
+        CSSPixels margins;
+        if (!has_horizontal_writing_mode)
+            margins = box_state.margin_left + box_state.margin_right;
+        else
+            margins = box_state.margin_top + box_state.margin_bottom;
```

Comment:
let's add and use here a getter for UsedValues that returns block/inline margin based on writing-mode

Reply:
I'm trying to organize this and at least what I'm coming up with is more complicated. Everything comes from `m_state.box_state` in `BlockFormattingContext` and I'm having to do a lot of redundant argument passing for things that are already computed in the scope of this function. I agree it would be nice to have a single call `auto pixels = get_pixels()` though. A lambda might be easier for now, especially if this isn't used very often

Reply:
I had in mind something like `box_state.get_block_margin_start(writing_mode)` and `box_state.get_inline_margin_end(writing_mode)`. it should require only passing writing mode as a parameter.

Reply:
okay I think I got something more reasonable using this approach. thinking it might also be good to VERIFY that size is positive after subtracting margins, but I can fix that up